test/xspec-variable.xspec

```diff
@@ -7,41 +7,230 @@
+				</x:scenario>
+			</x:scenario>
+		</x:scenario>
+	</x:scenario>
 
+	<x:scenario label="Variables can be redefined locally.">
+		<!-- Note: For x:variable elements that are direct children of x:description, expanded QNames must be unique.
+		 This test scenario is about *local* redefinitions of global or local variables. -->
+		<!-- For now, we assume any prefixes used in variable names are defined on x:description. -->
+		<!-- For a test that redefines a variable using URI-qualified notation, see xspec-eqname.xspec,
+		 which has xslt-version="3.0". -->
+		<x:scenario label="Local redefinition of global variable">
+			<x:variable name="myv:test-string" select="concat($myv:test-string,' (redefined locally using same prefix')" as="xs:string"/>
+			<x:variable name="myv_alt:test-string" select="concat($myv:test-string,' or different prefix)')" as="xs:string"/>
+			<x:call function="exactly-one">
+				<x:param select="$myv:test-string"/>
+			</x:call>
+			<x:expect label="yields correct redefined result."
+				select="'global (redefined locally using same prefix or different prefix)'"/>
+		</x:scenario>
+		<x:scenario label="Redefinition of variable inherited from outer scenario">
+			<x:variable name="myv:outer" select="('overwritten','data')" as="xs:string+"/>
+			<x:variable name="myv:outer" select="'value #1'" as="xs:string"/>
+			<x:scenario label="[inner]">
+				<x:variable name="myv:outer" select="concat($myv:outer,' (redefined locally using same prefix')" as="xs:string"/>
+				<x:variable name="myv_alt:outer" select="concat($myv:outer,' or different prefix)')" as="xs:string"/>
+				<x:call function="exactly-one">
+					<x:param select="$myv:outer"/>
+				</x:call>
+				<x:expect label="reflects value redefined in inner scenario."
+					select="'value #1 (redefined locally using same prefix or different prefix)'"/>
+			</x:scenario>
+		</x:scenario>
+		<x:scenario label="Redefinition of variable in same scenario">
+			<x:variable name="myv:local" select="('overwritten','data')" as="xs:string+"/>
+			<x:variable name="myv:local" select="'value #2'" as="xs:string"/>
+			<x:variable name="myv:local" select="concat($myv:local,' (redefined using same prefix')" as="xs:string"/>
+			<x:variable name="myv_alt:local" select="concat($myv:local,' or different prefix)')" as="xs:string"/>
+			<x:call function="exactly-one">
+				<x:param select="$myv:local"/>
+			</x:call>
+			<x:expect label="reflects value redefined within same scenario."
+				select="'value #2 (redefined using same prefix or different prefix)'"/>
+		</x:scenario>
+		<x:scenario label="The name 'result' in a non-XSpec namespace">
+			<x:call function="exactly-one">
+				<x:param select="'result of function call'"/>
+			</x:call>
+			<x:variable name="myv:result" select="'no conflict'" as="xs:string"/>
+			<x:expect label="has no conflict with $x:result" test="$myv:result eq 'no conflict'"/>
```

Comment:
Due to a tricky design of XSpec, this `x:expect[@test][empty(@select)][empty(child::node())]` is Success even when `$myv:result` is `()` (empty).
I fixed the other tests in #320.